Assets/Plugins/CountlySDK/Services/InitializationCountlyService.cs

```diff
@@ -52,21 +58,17 @@ internal async Task OnInitializationComplete()
                  */
                     await _locationService.SendIndependantLocationRequest();
                 }
-            }
-
-            if (!_configModel.EnableManualSessionHandling) {
+            } else if (!_configModel.EnableManualSessionHandling) {
                 //Start Session
                 await _sessionService.BeginSessionAsync();
             }
 
-            if (_configModel.EnableTestMode) {
+            if (_configModel.EnableTestMode || !_consentService.CheckConsent(Features.Push) || _configModel.NotificationMode == TestMode.None) {
```

Comment:
Rework this if so that there isn't a return statement inside it. That increases the future error chance of this function

Reply:
<img width="838" alt="Screenshot 2021-01-15 at 11 45 48 AM" src="https://user-images.githubusercontent.com/8924406/104690992-42251680-5727-11eb-896d-773df37b74dd.png">
it is the full picture.

Reply:
I know what the full picture looks like. It should still be reworked